content/intro-to-storybook/react/zh-TW/composite-component.md

```diff
@@ -164,87 +180,93 @@ function TaskList({ loading, tasks, onPinTask, onArchiveTask }) {
   ...
 }
 
-
-TaskList.propTypes = {
-  loading: PropTypes.bool,
-  tasks: PropTypes.arrayOf(Task.propTypes.task).isRequired,
-  onPinTask: PropTypes.func.isRequired,
-  onArchiveTask: PropTypes.func.isRequired,
-};
-
-TaskList.defaultProps = {
-  loading: false,
-};
-
-export default TaskList;
++ TaskList.propTypes = {
++  /** Checks if it's in loading state */
++  loading: PropTypes.bool,
++  /** The list of tasks */
++  tasks: PropTypes.arrayOf(Task.propTypes.task).isRequired,
++  /** Event to change the task to pinned */
++  onPinTask: PropTypes.func,
++  /** Event to change the task to archived */
++  onArchiveTask: PropTypes.func,
++ };
++ TaskList.defaultProps = {
++  loading: false,
++ };
 ```
 
-## 自動化測試
+## 自動測試
 
-在上一章中,我們學習瞭如何使用 Storyshots 快照測試 故事. `Task`測試沒有太多的複雜性,已然夠用了. 而`TaskList`增加了另一層複雜性,我們希望 以 自動測試 的方式驗證 某些輸入產生某些輸出. 為此,我們將使用建立單 元測試[jest-笑話](https://facebook.github.io/jest/)再加上測試渲染器等[Enzyme](http://airbnb.io/enzyme/).
+在上一章學到使用 Storyshot 為 story 進行快照測試。那時候，要測試 Task 渲染是否 OK 並不複雜。而 TaskList 增加另一層複雜度之後，就會想要相容自動測試，驗證特定輸入是否可以產生特定輸出。因此，要以 [Jest](https://facebook.github.io/jest/) 搭配測試渲染機制，建立單元測試。
 
 ![Jest logo](/intro-to-storybook/logo-jest.png)
 
-### 用 Jest 進行單元測試
+### 以 Jest 進行單元測試
 
-Storybook 故事 與 手動視覺化測試 和 快照測試 (見上文) 相結合,可以避免 UI 錯誤. 如果故事 涵蓋了 各種各樣的元件用例,並且我們使用的工具可以確保 人員檢查故事的任何變化,那麼錯誤的可能性就大大降低.
+Storybook 的 story、手動測試和快照測試已經能夠盡可能避免 UI 臭蟲。如果 story 涵蓋的元件使用情境已經廣泛，並且使用以人類進行檢查 story 變動的工具，錯誤就可能會比較少。
 
-然而,有時候魔鬼是在細節中. 需要一個明確有關這些細節的測試框架. 這讓我們進行了單元測試.
+然而，有時候魔鬼就是藏在細節裡，得要有讓細節顯而易見的測試框架，讓我們把目光放到單元測試。
 
-在我們的例子中,我們希望我們的`TaskList`,在傳遞 不固定 tasks 之前,呈現所有固定 tasks. 雖然我們有一個故事 (`withPinnedTasks`) 測試這個確切的場景; 但是如果元件停止對 這樣的任務 進行排序，那麼就人類看著來說，這可能是不明確的,_因為只看到表面與操作_, 這是一個 bug. 它肯定不會尖叫 **"錯誤!"** 直懟眼睛.
+現在的情況是，已經傳入 `tasks` 這個 props 的 `TaskList` 裡，將置頂任務在沒有置頂的**前面**渲染出來。即使已經有 `WithPinnedTasks` 這個 story，就是用來測試這情境。如果元件**不再**以如此方式排列任務，也就是出現臭蟲了，對於以人力來檢查來說，仍是模糊的。它絕對不會對大家的目光大喊**「出錯了」**！
 
-因此,為了避免這個問題,我們可以使用 Jest 將故事呈現給`DOM`,並執行一些`DOM`查詢程式碼,來驗證輸出的顯著特徵.
+因此，為了避免這問題，可以使用 Jest 來把 story 渲染至 DOM，然後執行一些 DOM 查詢程式碼，驗證結果的顯著特徵。story 格式的好處，是可以只要匯入測試裡的 story，然後就輸出了。
 
-建立一個名為的測試檔案`TaskList.test.js`. 在這裡,我們將構建我們的測試,對輸出進行斷言.
+新增名為 `src/components/TaskList.test.js` 的測試檔案。在這裡，要打造有明確結果的測試。
 
-```javascript
+```js:title=src/components/TaskList.test.js
 import React from 'react';
 import ReactDOM from 'react-dom';
-import TaskList from './TaskList';
-import { withPinnedTasks } from './TaskList.stories';
+
+import '@testing-library/jest-dom/extend-expect';
+
+import { WithPinnedTasks } from './TaskList.stories'; //👈  Our story imported here
 
 it('renders pinned tasks at the start of the list', () => {
   const div = document.createElement('div');
-  const events = { onPinTask: jest.fn(), onArchiveTask: jest.fn() };
-  ReactDOM.render(<TaskList tasks={withPinnedTasks} {...events} />, div);
+  //👇 Story 測試要用到的 args
+  ReactDOM.render(<WithPinnedTasks {...WithPinnedTasks.args} />, div);
 
-  // 我們期望首先渲染標題為“任務6（固定）”的任務，而不是最後
+  // 期望標題是 "Task 6 (pinned)" 的任務先渲染出來，而不是在尾端
```

Comment:
@ymcheung leaving this here but it applies to some other comments in the code snippets in the translation. If you don't mind leaving the snippets in their original form (English) we would appreciate it. I know that translating them would bring the "translation closer to home" (pardon the bad pun), but its something that has been applied to all translations currently available.